test/integration/controller/jobset_controller_test.go

```diff
@@ -257,6 +255,10 @@ var _ = ginkgo.Describe("JobSet controller", func() {
 					gomega.Eventually(util.CheckJobSetStatus, timeout, interval).WithArguments(ctx, k8sClient, js, up.expectedJobSetCondition).Should(gomega.Equal(true))
 				}
 			}
+			// Let's delete the jobset
+			gomega.Expect(k8sClient.Delete(ctx, js)).Should(gomega.Succeed())
```

Comment:
Same here, the util.DeleteNamespace() in AfterEach() should delete the jobset already, no need for this deletion here.